Clarify how to set up the lengthscale in the docs (#212)
JuliaGaussianProcesses/KernelFunctions.jl: docs/src/userguide.md

```julia
  k = SqExponentialKernel()
```
!!! tip "How do I set the lengthscale?" Instead of having lengthscale(s) for each kernel we use `Transform` objects (see [Transform](@ref)) which are directly going to act on the inputs before passing them to the kernel. 

> For example, if you want to to premultiply the input by 2.0, you can create your kernel with the following options:

Reviewer: ```suggestion
For example, if you want to premultiply the input by 2.0, you can create your kernel with the following options:
```

```julia
  k = transform(SqExponentialKernel(), 2.0)) # returns a TransformedKernel

>   k = TransformedKernel(SqExponentialKernel(), ScaleTransform(2.0))

Author: ```suggestion
```

>   k = TransformedKernel(SqExponentialKernel(), ScaleTransform(2.0))

Reviewer: I'm curious - is it useful to export `TransformedKernel` or would it be sufficient to only recommend `transform`?

Author: I know some people use it if they are more comfortable with having the right constructor name.

Reviewer: I don't mind to be honest. I was just curious why the documentation mentions two different ways here. In general, it is always possible to import `TransformedKernel` with `using KernelFunctions: TransformedKernel`, so it's more a design decision if we want to encourage users to use `transform` rather than `TransformedKernel`.

Author: That's fair, I will remove this line, anyway the docs are more complete in the `Transform` section

```

> In the example of the [SqExponentialKernel](@ref), you can reproduce the usual definition, $$\exp\left(-\frac{\|x-x'\|^2}{\rho^2}$$, by using the following `Transform` : `transform(SqExponentialKernel(), 1 / ρ`. 

Reviewer: ```suggestion
In the example of the [SqExponentialKernel](@ref), you can reproduce the usual definition, $$\exp\left(-\frac{\|x-x'\|^2}{\rho^2}\left)$$, by using `transform(SqExponentialKernel(), 1 / ρ)`. 
```

What if you want to differentiate through the kernel parameters? Even in a highly nested structure such as :
```julia
  k = transform(
        0.5 * SqExponentialKernel() * MaternKernel()
      + 0.2 * (transform(LinearKernel(), 2.0) + PolynomialKernel()),
      [0.1, 0.5])
```

> One can access the array of trainable parameters via `params` from `Flux.jl`

Reviewer: Does `params` return a flattened vector of parameters or a named tuple? Alternatively, one could use `first(Functors.functor(k))` to obtain a named tuple of parameters (Functors.functor is explained in more detail in the "Custom kernel" section).

Author: Well it actually would return a tuple of arrays to be exact. I think giving the Flux approach is probably more appropriate for first-timers. Functors is nice and light-weight but you kind of need to know what you are doing :D

Reviewer: Sure, most people will be unfamiliar with Functors and how to implement `Functors.functor` manually. But here it would be only about interpreting the output of `first(Functors.functor(k))` which is something such as
```julia
(x = 1, y = [1, 2, 3])
```
in the example in the [Functors README](https://github.com/FluxML/Functors.jl) and seems quite nice.

Author: Ok I will change it accordingly. Probably the best thing would be to give a straight example where the parameters can be optimized.

> One can access the array of trainable parameters via `params` from `Flux.jl`

Author: ```suggestion
One can access the named tuple of trainable parameters via `Functors.functor` from `Functors.jl`.
This mean that in practice you can implicitly optimize the kernel parameters by calling:
```julia
using Flux
hp = params(k)
gradient(hp) do
	... some loss function on the kernel ....
end
```

```
